src/withJWTProps.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import wrapDisplayName from 'recompose/wrapDisplayName'
+import jwtDecode from 'jwt-decode'
+
+export default (propName, propsMapping) => Target => {
+  class WithJWTProps extends Component {
+    constructor(props, context) {
+      super(props, context)
+
+      this.state = this.buildState(props[propName])
+    }
+
+    componentWillReceiveProps(nextProps) {
+      if (nextProps[propName] !== this.props[propName]) {
+        this.setState(this.buildState(nextProps[propName]))
+      }
+    }
+
+    buildState(token) {
+      try {
+        const decodedToken = jwtDecode(token)
```

Comment:
Also, if this HoC received a "decoder" as a parameter, this would be completely decoupled from the implementation of the decoder.

Something like:
```
decoder => token, mapToProps => component
```
where `decoder` is a function with the standard `token => err, data` signature.

Then you wouldn't need to pull in the dependency into this project.